src/org/rascalmpl/library/lang/rascal/tests/library/lang/java/m3/BasicM3Tests.rsc

```diff
@@ -6,152 +6,160 @@ import List;
 import Message;
 import IO;
 import util::FileSystem;
+import util::Math;
 import ValueIO;
 import Node;
 import String;
 import lang::java::m3::Core;
 import lang::java::m3::AST;
 
-private loc get(str path) = {l} := findResources(path) ? l : |not-found:///| + path;
-
-public loc unpackExampleProject(str name, loc projectZip) {
+@synopsis{Unpack a test source zip in a temporary folder for downstream analysis.}
+private loc unpackExampleProject(str name, loc projectZip) {
     targetRoot = |tmp:///<name>|;
     sourceRoot = projectZip[scheme = "jar+<projectZip.scheme>"][path = projectZip.path + "!/"];
```

Comment:
I think that we can actually use `zip+` now instead of `jar+`? At the time `zip` might not have been available?

Reply:
Yes I was also thinking that :-) But this private function is used for both unpacking jars and zips, and the jar unzipper does something special with the META-INF folder.